panel/widgets/select.py

```diff
@@ -147,47 +146,45 @@ def __init__(self, **params):
         super().__init__(**params)
         if self.size == 1:
             self.param.size.constant = True
+        self._validate_options_groups()
+
+    @param.depends('options', 'groups', watch=True)
```

Comment:
If we require param 1.12 we could start using `on_init`:

```suggestion

    @param.depends('options', 'groups', watch=True, on_init=True)
```